File: PR1/PR1_Ej3/HamburguesaNormalBuilder.py
from HamburguesaBuilder import HamburguesaBuilder
import time
class HamburguesaNormalBuilder(HamburguesaBuilder):
    def aniadePan(self):
        self.hamburguesa.pan = "Pan normal"
        time.sleep(0.1)

    def aniadeLechuga(self):
        self.hamburguesa.lechuga = "Lechuga fresca"
        time.sleep(0.1)

    def aniadeTomate(self):
        self.hamburguesa.tomate = "Tomate pera"
        time.sleep(0.1)

    def aniadeQuesoCabra(self):
        self.hamburguesa.quesoCabra = "Queso de cabra recién cortado"
        time.sleep(0.1)


    def aniadeCebolla(self):
        self.hamburguesa.cebolla = "Cebolla llorosa"
        time.sleep(0.1)


    def aniadePepinillos(self):
        self.hamburguesa.pepinillos = "Pepinillos"
        time.sleep(0.1)


    def aniadeBacon(self):
        self.hamburguesa.bacon = "Bacon grasiento"
        time.sleep(0.5)


    def aniadeCarne(self):
        self.hamburguesa.carne = "Carne Wagyu"
        time.sleep(0.8)

File: PR2/pantalla_pedidos_hamburgueseria/lib/HamburguesaNormalBuilder.dart
import 'HamburguesaBuilder.dart';
import 'Hamburguesa.dart';

class HamburguesaNormalBuilder extends HamburguesaBuilder {
  Hamburguesa hamburguesa = Hamburguesa() ;

  @override
  void aniadePan() {
    hamburguesa.pan = "Pan normal";
    sleep();
  }

  @override
  void aniadeLechuga() {
    hamburguesa.lechuga = "Lechuga fresca";
    sleep();
  }

  @override
  void aniadeTomate() {
    hamburguesa.tomate = "Tomate pera";
    sleep();
  }

  @override
  void aniadeQuesoCabra() {
    hamburguesa.quesoCabra = "Queso de cabra recién cortado";
    sleep();
  }

  @override
  void aniadeCebolla() {
    hamburguesa.cebolla = "Cebolla llorosa";
    sleep();
  }

  @override
  void aniadePepinillos() {
    hamburguesa.pepinillos = "Pepinillos";
    sleep();
  }

  @override
  void aniadeBacon() {
    hamburguesa.bacon = "Bacon grasiento";
    sleepLong();
  }

  @override
  void aniadeCarne() {
    hamburguesa.carne = "Carne Wagyu";
    sleepLong();
  }

  @override
  void aniadePrecio() {
    hamburguesa.precio = 5;
  }
}